cmd/gokr-packer/packer.go

```diff
@@ -646,36 +826,63 @@ func logic() error {
 		}
 	}
 
+	if *update == "yes" {
+		*update = "http://gokrazy:" + pw + "@" + *hostname + "/"
+	}
 	baseUrl, err := url.Parse(*update)
 	if err != nil {
 		return err
 	}
+
+	httpClient, foundMatchingCertificate, err := httpclient.GetTLSHttpClientByTLSFlag(useTLS, baseUrl)
+	// probe for https redirect, before sending credentials via http
+	probeResp, err := httpClient.Get("http://" + baseUrl.Host)
+	if err != nil {
+		return fmt.Errorf("probing url for https: %v", err)
+	}
+	if probeResp.TLS != nil {
```

Comment:
Instead of verifying that a TLS connection was established, we could also disable following redirects (see CheckRedirect in https://golang.org/pkg/net/http/#Client) and just test for the redirect HTTP response code. This saves one round-trip.